llvm/tools/llvm-cas/llvm-cas.cpp

```diff
@@ -457,11 +461,31 @@ static Expected<ObjectProxy> ingestFileSystemImpl(ObjectStore &CAS,
       });
 }
 
-int ingestFileSystem(ObjectStore &CAS, StringRef Path) {
+/// Check that we are not attempting to ingest the CAS into itself, which can
+/// accidentally create a weird or large cas.
+Error checkCASIngestPath(StringRef CASPath, StringRef DataPath) {
+  SmallString<128> RealCAS, RealData;
+  if (std::error_code EC = sys::fs::real_path(StringRef(CASPath), RealCAS))
+    return createFileError(CASPath, EC);
+  if (std::error_code EC = sys::fs::real_path(StringRef(DataPath), RealData))
+    return createFileError(CASPath, EC);
+  if (RealCAS.startswith(RealData) &&
```

Comment:
There is a `llvm::sys::path::starts_with` but I guess that one also doesn't support things like case insensitive system or the separator check.

Reply:
I don't see such a function in sys::path.  There is `static bool starts_with` in Path.cpp, but it is just doing `StringRef::starts_with` if it's a unix path.  This probably *should* be an API though.